src/structures/Channel.js

```diff
@@ -90,15 +92,26 @@ class Channel extends Base {
    * @returns {Promise<Channel>}
    */
   fetch() {
+    if (this.type === 'group') return Promise.reject(new Error('FETCH_GROUP_DM_CHANNEL'));
```

Comment:
Override the method in the deriving class (`PartialGroupDMChannel`) and throw the error there.